corehq/apps/userreports/views.py

```diff
@@ -439,6 +441,28 @@ def preview_data_source(request, domain, config_id):
     return render(request, "userreports/preview_data.html", context)
 
 
+def process_url_params(params, column_descriptions):
+    columns = {column['name']: column for column in column_descriptions}
+    keyword_filters = {}
+    sql_filters = []
+    for key, value in params.items():
+        if key == 'format':
+            continue
+        column = columns.get(key, None)
+        if not column:
+            raise BadSpecError('Invalid filter parameter: {}'.format(key))
+
+        if (
+            value == 'last30'
+            and isinstance(column['type'], (types.Date, types.DateTime))
+        ):
+            expr = column['expr']
+            sql_filters.append(expr.between('2014-02-02', '2014-10-02'))
+        else:
+            keyword_filters[key] = value
+    return keyword_filters, sql_filters
```

Comment:
I would return an object here that you can use to encapsulate the query logic somehow, possibly just a `namedtuple` that gets a `format` (?), `keyword_filters` and `sql_filters`... though actually if you make it a full-blown class then you can abstract out the `sql_filters` to a function call (let me know if that doesn't make sense and I can quickly sketch up what i mean).